Scalar.Common/Git/GitProcess.cs

```diff
@@ -604,6 +604,65 @@ public Result RemoteAdd(string remoteName, string url)
             return this.InvokeGitAgainstDotGitFolder("remote add " + remoteName + " " + url);
         }
 
+        public bool TryGetRemoteDefaultBranch(string remoteName, out string defaultBranch, out string error)
+        {
+            defaultBranch = null;
+            error = null;
+
+            Result lsRemoteResult = this.InvokeGitAgainstDotGitFolder("ls-remote --symref origin HEAD");
+            if (lsRemoteResult.ExitCodeIsFailure)
+            {
+                error = "Failed to call ls-remote to determine remote HEAD";
+                return false;
+            }
+
+            string output = lsRemoteResult.Output;
+
+            const string refPrefix = "ref: ";
+            const string headsPrefix = "ref: refs/heads/";
+            const string suffix = "\tHEAD";
+
+            var cmp = StringComparison.Ordinal;
+            foreach (string line in output.Split('\n'))
+            {
+                int suffixStart = line.IndexOf(suffix, cmp);
+                if (line.StartsWith(refPrefix, cmp) && suffixStart > -1)
+                {
+                    // HEAD is a branch
+                    int prefixStart = line.IndexOf(headsPrefix, cmp);
+                    int prefixEnd = prefixStart + headsPrefix.Length;
+                    if (prefixStart > -1 && prefixEnd < suffixStart)
+                    {
+                        defaultBranch = line.Substring(prefixEnd, suffixStart - prefixEnd);
```

Comment:
This should probably be `line.Substring(headsPrefix.Length, suffixStart - headsPrefix.Length)`.